libs/LedKit/source/animations/BlinkGreen.cpp

```diff
@@ -17,6 +17,11 @@ void BlinkGreen::setLeds(interface::LED &ears, interface::LED &belt)
 	_belt = &belt;
 }
 
+auto BlinkGreen::hasEnded() -> bool
+{
+	return _ended;
+}
```

Comment:
par rapport à mon commentaire plus haut, la sémantique de hasEnded et de isStopped n'est pas la même.

d'abord, il faudrait changer `hasEnded` par `didEnd` c'est plus simple et plus concret en anglais. le has/have + verbe au passé (present perfect) signifie que ça a eu lieu dans le passé et que c'est toujours le cas dans le présent, comme dans "I have waited for 5 hours"

du coup ça voudrait dire que l'animation a commencé à finir dans le passé et est toujours entrain de se finir maintenant.

comme c'est une action ponctuelle, c'est le didEnd qu'il faut utiliser.

did + verb permet de marquer le fait que c'est dans le passé. par exemple:

- are you sure the war ended in 1944?
- no, the war **did** end in 1945 == no, the war ended in 1945

l'intérêt de didVerb, c'est qu'il ne faut pas réfléchir à la forme past du verb utilisé. 

dans iOS, ils ont les deux formes pour avant / après : viewWillAppear, viewDidAppear -- à noter que "pendant" n'est pas "possible" parce que par définition c'est ponctuel, donc tu peux pas le faire en même temps, seulement juste avant ou juste après

maintenant la différence entre didEnd, didStop et isStopped

- didEnd - ça s'est terminé
- didStop - ça a été arrêté -- les deux peuvent être équivalent, mais stop rajoute une notion d'intervention extérieur ou de changement brusque avant la fin, end c'est vraiment qu'on est au bout, il n'y a plus rien à faire
- isStopped - c'est actuellement arrêté. on s'est pas si ça a été en marche à un moment, c'est un état actuel qui ne préfigure pas de l'ordre des actions

a noté que is ended n'est pas très anglais, on dira plutôt "is done" or "is finished" or "is adjourned"

donc il faut réfléchir à ce qu'on veut dire avec la fonction : est-ce qu'on parle de l'état/status en cours, dans ce cas là on a `isStopped` ou `isNotRunning` ou `isIdle` 

ou est-ce qu'on parle de ce qui se passe avec une notion de temporalité, dans ce cas là on aura `didEnd` ou `didStop` -- mais retourner un bool dans ce cas, c'est étrange, on voudra plutôt passer un callback à executer quand on arrive à ce moment là ou alors définir une fonction qui sera appelé quand l'animation aura terminé ou sera arrêtée